File: credit/credit.py
# prompt user for input

while True:
    num_string = input("Number: ")
    if num_string.isdigit():
        break
    else:
        continue

num_int = int(num_string)

# add the checksums
sum1 = 0
sum2 = 0

for i in range(0, len(num_string), 2):
    sum2 += num_int % 10
    num_int = num_int // 10

    temp = (num_int % 10) * 2
    if temp < 10:
        sum1 += temp
    else:
        sum1 += (temp % 10) + (temp // 10)

    num_int = num_int // 10


final_sum = sum1 + sum2

# check the validity

if final_sum % 10 == 0:
    if len(num_string) == 15 and num_string[0] == '3' and (num_string[1] == '4' or num_string[1] == '7'):
        print("AMEX")
    elif len(num_string) == 16 and int(num_string[0]) == 5 and (int(num_string[1]) >= 1 and int(num_string[1]) <= 5):
        print("MASTERCARD")
    elif (len(num_string) == 16 or len(num_string) == 13) and int(num_string[0]) == 4:
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long number;

    number = get_long("Number: ");

    int a, b, c, d;

    int sum_1 = 0;
    int sum_2 = 0;
    int digits = 0;

    long n = number;
    long x = number;

    do
    {
        digits++;
        x /= 10;
    }
    while (x != 0);

    for (int i = 0; i < digits; i++)
    {
        a = n % 10;
        sum_2 = sum_2 + a;
        n /= 10;
        b = n % 10;
        n /= 10;
        c = b * 2;

        if (c < 10)
        {
            sum_1 = sum_1 + c;
        }
        else
        {
            d = c % 10;
            c /= 10;
            sum_1 = sum_1 + c + d;
        }
    }

    if (((sum_1 + sum_2)) % 10 == 0)
    {
        if ((digits == 15) && ((number / 10000000000000 == 34) || (number / 10000000000000 == 37)))
        {
            printf("AMEX\n");
        }

        else if ((digits == 16) && ((number / 100000000000000 >= 51) && (number / 100000000000000 <= 55)))
        {
            printf("MASTERCARD\n");
        }

        else if (((digits == 13) || (digits == 16)) && ((number / 1000000000000 == 4) || (number / 1000000000000000 == 4)))
        {
            printf("VISA\n");
        }

        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}